tools/FileTestDefs.sh

```diff
@@ -414,6 +414,26 @@ namespace_details_test() {
 }
 standard_checks+=(namespace_details)
 
+# Check for .cpp includes in headers and tpp files
+cpp_includes_in_includibles() {
+    is_includible "$1" && staged_grep -q "include .*\.cpp" $1
```

Comment:
Is there a reason to allow this in cpp files?

Reply:
It seems "less wrong" but I think with `tpp` files we have a design that should not require us to ever include a cpp, right?

Reply:
Yeah.  Currently, there is one such include in `tests/Unit/DataStructures/Test_OrientVariablesOnSlice.cpp`, which is a bug, and a bunch of includes of `Parallel/CharmMain.cpp`, which would have to be dealt with.

So I guess our choices are: (1) rename CharmMain.cpp to a tpp or something, (2) whitelist CharmMain.cpp includes, or (3) leave this check as is for now and decide what to do with the cpp case later because it will get Travis fixed faster.  I'm fine with any of these.

Reply:
I think for the `ChairmMain.cpp` I'll go with `tpp` for now. While not "techincally consistent" (it doesn't contain template definitions) it should be used in basically the same way.